Adds template-file rendering examples in the bug report test-template for ActionView

shivabhusal opened the pull request:
## Motivation

When developers report view-template rendering issues, they often struggle to provide reproducible code. As a result, many issue descriptions are unclear or incomplete, making them difficult to understand or reproduce. This is largely because we do not currently provide a predefined template for such reports.

## Details

This PR enhances the existing bug report test template for Action View by adding clearer guidance and structure for issues related to template file rendering. The updated template helps contributors create executable, reproducible test cases, improving the quality and efficiency of issue triage and resolution.

shivabhusal commented:
@zzak Here are a few example issues that I believe would be clearer with the new reproducible template:

- https://github.com/rails/rails/issues/56239
- https://github.com/rails/rails/issues/45636
- https://github.com/rails/rails/issues/14991
- https://github.com/rails/rails/issues/55161
- https://github.com/rails/rails/issues/42201

Happy to iterate further. Suggestions to make the template even more useful are very welcome.

zzak commented:
At that point, why not just make this the default for AV reports?

seanpdoyle commented:
Are bug report templates executed as part of CI? IF so, should the PR description remove the `[ci skip]` suffix so that these proposed changes are evaluated and pass before merge?

shivabhusal committed:
Adds template-file rendering examples in the bug report test-template for ActionView

- It adds tests to validate rendering of templates & partials
- Adds ActionView tests for template and inline rendering with titleize helper
- Uses `Pathname.new` for improved path handling
- Apply suggestion from @seanpdoyle
- Adds setup and teardown methods for clarity and flexibility

Co-authored-by: Sean Doyle <seanpdoyle@users.noreply.github.com>

zzak commented on guides/bug_report_templates/action_view.rb:
```diff
@@ -16,26 +16,62 @@
+  end
+
+
+  test "Action View inline rendering" do
```
```suggestion
  def test_action_view_render_inline
```

zzak commented:
Can you please [squash your commits](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits) and I think the title/description should call out we're _expanding_ the Action View bug report to include template file rendering.
